compile/integration/src/main/scala/sbt/compiler/AggressiveCompile.scala

```diff
@@ -149,9 +149,12 @@ class AggressiveCompile(cacheFile: File) {
           // previous Analysis completely and start with empty Analysis object
           // that supports the particular value of the `nameHashing` flag.
           // Otherwise we'll be getting UnsupportedOperationExceptions
+          log.warn("Ignoring previous analysis due to incompatible nameHashing setting.")
           Analysis.empty(currentSetup.nameHashing)
         case Some(previous) if equiv.equiv(previous, currentSetup) => previousAnalysis
-        case _ => Incremental.prune(sourcesSet, previousAnalysis)
+        case _ =>
+          log.warn("Pruning sources from previous analysis, due to incompatible CompileSetup.")
+          Incremental.prune(sourcesSet, previousAnalysis)
```

Comment:
This is actually the one that bites us the most. Changes to CompileSetup happen e.g., when we upgrade a scalac plugin or change memory options. It's not at all obvious that this will invalidate your analysis and trigger a full compile, so we think it's worth a warning.